src/SSHDebugTests/SSHConnectionStringTests.cs

```diff
@@ -143,6 +144,15 @@ public void Ipv4ConnectionStrings()
                     expectedHostname = "10.10.10.10",
                     expectedPort = ConnectionManager.DefaultSSHPort
                 });
+            ipv4TestStrings.Add(
+                 new ConnectionStringTestItem()
+                 {
+                     // valid username no port with password
+                     rawConnectionString = "user:pass@10.10.10.10",
+                     expectedUsername = "user",
+                     expectedHostname = "10.10.10.10",
+                     expectedPort = ConnectionManager.DefaultSSHPort
```

Comment:
>expectedPort [](http://example.com/codeflow?start=21&length=12)

Should we add expected password testing also?

Reply:
I started too, but given the secure string I can't check the value. I can check that the expected password is present?

Reply:
You can convert a SecureString to a string with this:

```
            IntPtr bstr = System.Runtime.InteropServices.Marshal.SecureStringToBSTR(s);
            string value = System.Runtime.InteropServices.Marshal.PtrToStringBSTR(bstr);
            System.Runtime.InteropServices.Marshal.FreeBSTR(bstr);
```

Reply:
Excellent thanks. Added the password checking.